ipa-core/src/report/hybrid.rs

```diff
@@ -1,18 +1,79 @@
-use std::{collections::HashSet, ops::Add};
+use std::{
+    collections::HashSet,
+    fmt::{Display, Formatter},
+    marker::PhantomData,
+    ops::{Add, Deref},
+};
 
-use bytes::Bytes;
-use generic_array::ArrayLength;
+use bytes::{BufMut, Bytes};
+use generic_array::{ArrayLength, GenericArray};
+use hpke::Serializable as _;
 use rand_core::{CryptoRng, RngCore};
 use typenum::{Sum, Unsigned, U16};
 
 use crate::{
-    error::Error,
+    error::{BoxError, Error},
     ff::{boolean_array::BA64, Serializable},
-    hpke::{EncapsulationSize, PrivateKeyRegistry, PublicKeyRegistry, TagSize},
-    report::{EncryptedOprfReport, EventType, InvalidReportError, KeyIdentifier},
+    hpke::{
+        hybrid_open_in_place, hybrid_seal_in_place, CryptError, EncapsulationSize,
+        PrivateKeyRegistry, PublicKeyRegistry, TagSize,
+    },
+    report::{
+        hybrid_info::HybridImpressionInfo, EncryptedOprfReport, EventType, InvalidReportError,
+        KeyIdentifier,
+    },
     secret_sharing::{replicated::semi_honest::AdditiveShare as Replicated, SharedValue},
 };
 
+// TODO(679): This needs to come from configuration.
+static HELPER_ORIGIN: &str = "github.com/private-attribution";
+
+#[derive(Debug)]
+pub struct NonAsciiStringError {
```

Comment:
Can you add a few tests for this error? Specially for the cases dealing with non ascii and how does it handle them?

What happens if the input isn't even utf8?

Reply:
I added a non-ascii test, but it looks like rust doesn't even support strings that aren't utf-8

Reply:
Right, Strings in rust they are utf-8. If the error can happen on arbitrary `[u8]`... do you need to make`[u8]` your `input` instead of `String` in order to capture what was the reason of the error?

In other words, this error is just for non ascii or for any non-conforming string?

Reply:
Ohhh, yeah I see the problem now. Good catch. I just removed from [u8] capability. Requiring a string should guarantee that it's utf-8

Reply:
It doesn't seem enough to me to require UTF-8. Any encoding that requires 2 bytes or more would not be a valid ASCII, so we still need to check `is_ascii`

Reply:
I'm confused, where do you want to add the `is_ascii` check? The function that can throw the NonAsciiStringError (the new() function for HybridImpressionInfo) calls `is_ascii` and throws this error if not. But the error has the option to print the non-ascii string, so I don't think adding a `is_ascii` check there makes sense.

Reply:
yea, I just wanted to clarify that we can't simply check for UTF8 and be done with it. The `is_ascii` check still needs to be in place in order for this to be correct